src/shared/views/RecommendedCards.js

```diff
@@ -77,13 +78,92 @@ type OrderedProject = {
     navigation: NavigationProp,
     projects: Array<OrderedProject>,
 };
 
+// request only active projects from firebase (status === 'active')
+// firebase doesn't allow multiple query params, so for project types we filter in render()
+// but here we can still limit to 20 projects maximum
+// `path` defines where the resulting data is copied in the redux store
+// (state.firebase.ordered.projects in this case, because we've asked for `orderByChild`)
+const projectsQuery = {
+    isQuery: true,
+    path: 'v2/projects',
+    queryId: 'projectsQuery',
+    queryParams: ['orderByChild=status', 'equalTo=active', 'limitToFirst=20'],
+    storeAs: 'projects',
+    type: 'value',
+};
+// load any announcement data from firebase
+// (state.firebase.data.announcement here because we've not ordered the query)
+const announcementQuery = {
+    isQuery: true,
+    path: 'v2/announcement',
+    queryId: 'announcementQuery',
+    queryParams: ['limitToLast=2'],
+    storeAs: 'announcement',
+    type: 'value',
+};
+
 class _RecommendedCards extends React.Component<Props> {
     tutorialModal: ?Modal;
 
+    willBlurAnnouncementSubscription: NavigationEventSubscription;
+
+    willFocusAnnouncementSubscription: NavigationEventSubscription;
+
+    willBlurProjectSubscription: NavigationEventSubscription;
+
+    willFocusProjectSubscription: NavigationEventSubscription;
+
+    componentDidMount() {
+        this.subscribeToProjects();
+        this.subscribeToAnnouncements();
+    }
+
+    componentWillUnmount() {
+        this.willFocusAnnouncementSubscription.remove();
+        this.willBlurAnnouncementSubscription.remove();
+        this.willFocusProjectSubscription.remove();
+        this.willBlurProjectSubscription.remove();
+    }
+
+    subscribeToAnnouncements = () => {
+        const { type, path, storeAs, ...options } = announcementQuery;
+        const { firebase, navigation } = this.props;
+        this.willFocusAnnouncementSubscription = navigation.addListener(
+            'willFocus',
+            () => {
+                firebase.watchEvent(type, path, storeAs, options);
+            },
+        );
+        this.willBlurAnnouncementSubscription = navigation.addListener(
+            'willBlur',
+            () => {
+                firebase.unWatchEvent(type, path, storeAs, options);
+            },
+        );
+    };
+
+    subscribeToProjects = () => {
+        const { type, path, storeAs, ...options } = projectsQuery;
+        const { firebase, navigation } = this.props;
+        this.willFocusProjectSubscription = navigation.addListener(
+            'willFocus',
+            () => {
+                firebase.watchEvent(type, path, storeAs, options);
+            },
+        );
+        this.willBlurProjectSubscription = navigation.addListener(
+            'willBlur',
+            () => {
+                firebase.unWatchEvent(type, path, storeAs, options);
+            },
+        );
+    };
+
     closeModal3 = () => {
```

Comment:
which element is `Modal3`. Is there a better name?

Reply:
I am not sure, but as I look around the code, it's pretty clear that I need to spend some time renaming variable, functions and classes, the current names are rather confusing...